core/include/drivers/semihosting_console.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2024 Andes Technology Corporation
+ */
+#ifndef __DRIVERS_SEMIHOSTING_CONSOLE_H
+#define __DRIVERS_SEMIHOSTING_CONSOLE_H
+
+#include <drivers/serial.h>
+
+/*
+ * struct semihosting_console_data - Structure for semihosting console driver
+ * @chip - General structure for each serial chip
+ * @file_path - Path of the file on the semihosting host system or NULL if using
+ *              semihosting console
+ * @fd - Handle of the file at @file_path or -1 if using the semihosting console
+ */
+struct semihosting_console_data {
+	struct serial_chip chip;
+	const char *file_path;
```

Comment:
`file_path` is not needed it seems (set but never read)

Reply:
Yes, `fd` is enough. I can remove `file_path`.